arangod/Aql/JoinExecutor.cpp

```diff
@@ -37,38 +37,73 @@ JoinExecutor::JoinExecutor(Fetcher& fetcher, Infos& infos)
   constructStrategy();
 }
 
+namespace {
+struct SpanCoveringData : arangodb::IndexIteratorCoveringData {
+  explicit SpanCoveringData(std::span<VPackSlice> data) : _data(data) {}
+  bool isArray() const noexcept override { return true; }
+  VPackSlice at(size_t i) const override { return _data[i]; }
+  arangodb::velocypack::ValueLength length() const override {
+    return _data.size();
+  }
+
+  std::span<VPackSlice> _data;
+};
+}  // namespace
+
 auto JoinExecutor::produceRows(AqlItemBlockInputRange& inputRange,
                                OutputAqlItemRow& output)
     -> std::tuple<ExecutorState, Stats, AqlCall> {
+  VPackBuilder projectionsBuilder;
```

Comment:
Creating VPackBuilders repeatedly for every invocation of `produceRows()` seems expensive, even if the method is only called every 1000 rows or so.
VPackBuilders are quite heavy underneath, so I suggest that we create a VPackBuilder instance variable instead, which we can recycle via `VPackBuilder::clear()` upon every iteration.